rust/src/smb/smb.rs

```diff
@@ -53,6 +53,82 @@ use crate::smb::events::*;
 use crate::smb::files::*;
 use crate::smb::smb2_ioctl::*;
 
+#[repr(C)]
+pub enum SMBFrameType {
+    NBSS = 0,
+    NBSSHdr = 1,
+    NBSSData = 2,
```

Comment:
I think its time to make a derive macro for this, I don't see a way to do it purely with generics..  I'm thinking something like:
```
#[repr(C)]
#[derive(AppLayerFrame)]
pub enum SMBFramtType {
    #[frame(name = "nbss.pdu")
    NBSS,
    #[frame(name = "nbss.hdr")
    NBSSHdr,
}
```

We could get rid of the name by having a transformation method, like applayerevents use..

NbssPdu => "nbss.pdu"
Smb3Hdr => "smb3.hdr"

And do we really need the repr(C) here? It looks like we convert to u8 before passing over FFI in all cases.